build/build-static-binaries.sh

```diff
@@ -12,17 +12,26 @@ source $(dirname $0)/build-common.sh
 # commands in the if-branch to be executed within the docker
 # container.
 if [ "${1-}" = "docker" ]; then
+    test_build_dir=$(mktemp -d test-binaries.XXXX)
     time make STATIC=1 build
-    time make STATIC=1 testbuild PKG=./sql
-    time make STATIC=1 testbuild PKG=./acceptance TAGS=acceptance
+    # sql/sql.test is built standalone as well to simplify the nightly logictest runs.
+    time make STATIC=1 testbuild STRIPPED=1 PKG=./sql
+    time make STATIC=1 testbuild STRIPPED=1 PKG=./acceptance TAGS=acceptance
+    time make STATIC=1 testbuild STRIPPED=1 DIR=${test_build_dir}
 
+    # We don't check all test binaries, but one from each invocation.
     check_static cockroach
     check_static sql/sql.test
     check_static acceptance/acceptance.test
+    check_static ${test_build_dir}/github.com/cockroachdb/cockroach/sql/sql.test
 
     strip -S cockroach
-    strip -S sql/sql.test
-    strip -S acceptance/acceptance.test
+
+    rm -f static-tests.tar.gz
+    # Skip the project/repo part of the path inside the tarball.
+    # Even for stripped binaries, gzip results in 167MB (21s) vs 512MB (3s).
```

Comment:
sorry, this isn't particularly clear. this `tarball size (time spent building)` with gzip and without respectively.